test_app/authentication/service_token_auth.py

```diff
@@ -0,0 +1,39 @@
+import jwt
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework.authentication import BaseAuthentication, get_authorization_header
+
+from ansible_base.resource_registry.resource_server import get_resource_server_config
+
+User = get_user_model()
+
+
+class ServiceTokenAuthentication(BaseAuthentication):
+    keyword = "Token"
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+
+        if not auth or auth[0].lower() != self.keyword.lower().encode():
+            return None
```

Comment:
If I am reading this right, the `or`, is just checking for non-UTF8 characters in the authorization header with the JWT is that correct? If so, can we add a comment as to why we are doing that? Also, if this is for security (like an injection) should we should we have a function that wraps `get_authorization_header` and does this so we are protected everywhere?